daml-lf/transaction-test-lib/src/main/scala/lf/value/test/ValueGenerators.scala

```diff
@@ -84,7 +84,7 @@ object ValueGenerators {
       firstChars ++ "1234567890"
     for {
       h <- Gen.oneOf(firstChars)
-      t <- Gen.listOf(Gen.oneOf(mainChars))
+      t <- Gen.listOfN(999, Gen.oneOf(mainChars)) //Name is max 999 characters
```

Comment:
This has to be limited to 999 as otherwise generator generates invalid name:
`Message: Name is too long (max: 1000)`

This can be reproduced by playing around with `minSize` of `PropertyCheckConfiguration` of tests using these generators.

Reply:
I don't think we want exactly 1000 characters in every name, though, as `listOfN` will deliver; perhaps we could `choose` a size (with a bias towards small names, perhaps; we don't get big value out of big names in most tests I'd wager).

Reply:
Thanks for pointing it out - I missed it somehow. Not sure yet how to make it biased towards small names, suggestions? probability distribution?

Reply:
I'd just use `frequency` to pick the second (upper-bound) parameter to `choose`. So you could have 90% small names, 8% small-to-medium names, 2% any size.